### PR TITLE
RAFT: Replace slog with logrus for logs consistency 

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:
combined effort with @trengrj 

- replaces all `slog` usage with logrus  
- implements the `hclog.Logger` interface as a a wrapper for logrus to be used in RAFT as required for the config logger

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: adapters/handlers/rest/configure_api.go [hunk 1]
```diff
@@ -650,28 +649,28 @@ func logger() *logrus.Logger {
...
-	var handler slog.Handler
-	if logTextFormat() {
-		handler = slog.NewTextHandler(os.Stderr, &opts)
-	} else {
-		handler = slog.NewJSONHandler(os.Stderr, &opts)
-	}
-
-	return slog.New(handler)
-}
+// func sLogger() *slog.Logger {
```
Comment (reyreaud-l) [inline adapters/handlers/rest/configure_api.go:652]: thoughts(non-blocker): we should just delete dead code unless there is a strong reason to keep it around

File: cluster/store/logger.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+}
+
+type hclogLogrus struct {
+	entry *logrus.Entry
+	name  string
+}
+
+func (hclogger *hclogLogrus) GetLevel() hclog.Level {
+	switch hclogger.entry.Logger.Level {
```
Comment (reyreaud-l) [inline cluster/store/logger.go:37]: must-fix: lint failure we're missing cases

File: cluster/store/logger.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+	case logrus.ErrorLevel:
+		return hclog.Error
+	case logrus.FatalLevel:
+		return hclog.Error
+	}
+	return hclog.DefaultLevel
+}
+
+func (hclogger *hclogLogrus) Log(level hclog.Level, msg string, args ...interface{}) {
+	switch level {
```
Comment (reyreaud-l) [inline cluster/store/logger.go:55]: Same here as above

**Review by reyreaud-l — CHANGES_REQUESTED** [review]
Looks good, requested change on the switch I think we should cover all cases. Let me know if you disagree and we can talk about it and ignore it in the linter if we feel this is the right move to make.

File: adapters/handlers/rest/configure_api.go (hunk 1 above)
Comment (moogacs) [reply adapters/handlers/rest/configure_api.go:652]: Agreed

**Comment by moogacs** [comment]
ageed in all points, just missed the linter

File: cluster/store/logger.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	case logrus.TraceLevel:
+		return hclog.Trace
+	case logrus.DebugLevel:
+		return hclog.Debug
+	case logrus.InfoLevel:
+		return hclog.Info
+	case logrus.WarnLevel:
+		return hclog.Warn
+	case logrus.ErrorLevel:
+	case logrus.FatalLevel:
```
Comment (reyreaud-l) [inline cluster/store/logger.go:47]: question: I always assumed logging fatal would kill/panic the process. I feel like this change breaks that assumption. Therefore do we log Fatal anywhere and should update that code to panic/exit afterward ?

Comment (moogacs) [reply cluster/store/logger.go:47]: actually i thin you'r right w.r.t https://github.com/sirupsen/logrus/blob/dd1b4c2e81afc5c255f216a722b012ed26be57df/hooks/syslog/syslog.go#L37C1-L39C32 

i may need to change fatal and panic to panic

Comment (moogacs) [reply cluster/store/logger.go:47]: the problem is now that hclog doesn't have the same level :D

Comment (moogacs) [reply cluster/store/logger.go:47]: well given that also that logrus<->gclog they don't have the same leveling and to [follow docs in hclog](https://github.com/hashicorp/go-hclog/blob/main/logger.go#L43)  it seems that is the max. we can do as level point of view, hclog won't panic that is the diff

Comment (reyreaud-l) [reply cluster/store/logger.go:47]: That sounds OK to me

**Review by reyreaud-l — APPROVED** [review]

**Pull request merged**